test/Assertion.test.js

```diff
@@ -28,6 +28,21 @@ describe("assertion", () => {
         expect(assertion.evaluate(obj)).toBe(true);
     });
 
+    test("evaluate array", () => {
+        const obj = { val: "Here is a test +" };
+        let assertion = new Assertion("val", "=~", ["/.*/", "not a test"]);
+        expect(assertion.evaluate(obj)).toBe(true);
+
+        assertion = new Assertion("val", "==", ["/Not here.*/", "Here is a test +"]);
+        expect(assertion.evaluate(obj)).toBe(true);
+
+        assertion = new Assertion("val", "==", ["/.*/", "Here does not match"]);
+        expect(assertion.evaluate(obj)).toBe(true);
+
+        assertion = new Assertion("val", "==", ["Not match", "Here does not match"]);
+        expect(assertion.evaluate(obj)).toBe(false);
+    });
+
```

Comment:
this could be written as:

```
describe('evaluate array', () => {
    const obj = { val: "Here is a test +" };
    [
      { value: 'val', operator:'=~', list: ["/.*/", "not a test"], expectedValue: true },
      { value: 'val', operator:'==', list: ["/Not here.*/", "Here is a test +"]), expectedValue: true },
      { value: 'val', operator:'==', list: ["Not match", "Here does not match"], expectedValue: false },
    ].forEach(({ value, operator, list, expectedValue }) =>
      it('should be expectedValue', () => {
        assertion = new Assertion(value, operator, list, );
        expect(assertion.evaluate(obj)).toBe(expectedValue);
      }),
    );
  });
```

Reply:
I don't like writing tests like that because if it fails and I jump to the line of failure, it's not immediately clear what happened. I know it can still be figured out, but making it as simple as possible I believe works better.